service/smithyprototype/lexruntimeservice/api_op_PutSession.go

```diff
@@ -389,3 +394,103 @@ type PutSessionResponse struct {
 func (r *PutSessionResponse) SDKResponseMetdata() *aws.Response {
 	return r.response
 }
+
+type putSessionSerializeMiddleware struct{}
+
+// ID is the middleware identifier
+func (p putSessionSerializeMiddleware) ID() string {
+	return "PutSessionSerializeMiddleware"
+}
+
+func (p putSessionSerializeMiddleware) HandleSerialize(ctx context.Context, in middleware.SerializeInput, next middleware.SerializeHandler) (
+	out middleware.SerializeOutput, metadata middleware.Metadata, err error,
+) {
+	request, ok := in.Request.(*smithyHTTP.Request)
+	if !ok {
+		return out, metadata, fmt.Errorf("unknown transport type %T", in.Request)
+	}
+
+	input, ok := in.Parameters.(*PutSessionInput)
+	if !ok {
+		return out, metadata, fmt.Errorf("unknown input parameters type %T", in.Parameters)
+	}
+
+	restEncoder := rest.NewEncoder(request.Request)
+
+	restEncoder.AddHeader("Content-Type").String("application/json")
+
+	if err := serializePutSessionInputAWSREST(input, restEncoder); err != nil {
```

Comment:
Should serializing functions be in a separate go file rater than being in operation file? (That was our initial design for serializers)